packages/block-library/src/paragraph/edit.js

```diff
@@ -176,12 +128,33 @@ function ParagraphBlock( {
 
 	const richTextStyles = useMemo(
 		() => ( {
-			backgroundColor: backgroundColor.color,
-			color: textColor.color,
 			fontSize: fontSize.size ? fontSize.size + 'px' : undefined,
 			direction,
 		} ),
-		[ backgroundColor.color, textColor.color, fontSize.size, direction ]
+		[ fontSize.size, direction ]
+	);
+
+	const { TextColor, BackgroundColor, InspectorControlsColorPanel } = __experimentalUseColors(
+		[
+			{ name: 'textColor', property: 'color' },
+			{ name: 'backgroundColor', className: 'has-background' },
+		],
+		{
+			panelChildren: ( components ) => {
```

Comment:
I don't like this that much, makes me wonder, why this is not just an option like `contrastChecker: true` or something like that. (a more declarative thing)

(Not a blocker for this PR but more a thought about the API useColors)

Reply:
https://github.com/WordPress/gutenberg/pull/18148#discussion_r339819339

We could expand the contrast checker configuration options. cc @jorgefilipecosta 

Like this:

```js
const { TextColor, BackgroundColor, InspectorControlsColorPanel } = __experimentalUseColors(
		[
			{ name: 'textColor', property: 'color' },
			{ name: 'backgroundColor', className: 'has-background' },
		],
		{
				contrastCheckers: [{ textColor: 'white' }], // Single/multiple with literal.
				contrastCheckers: 'white', // All with literal.
				contrastCheckers: [{ textColor: ({ BackgroundColor: { value } }) => value }], // Single/multiple with dependency.
				contrastCheckers: ({ BackgroundColor: { value } }) => value // All with dependency.
				contrastCheckers: [{ textColor: true }], // Single/multiple with DOM parent background resolution.
				contrastCheckers: true, // All with DOM parent background resolution.
		}
);
```

What do you guys think?

Reply:
I personally don't know enough about these features (contrast checkers) to decide if it's a great API or not but from a consumer's side, it seems reasonable and something I can learn/understand.

My question is do we need all these variations? Can we start with the simplest ones and add variations as we need them?

Reply:
We already support `// All with literal.`, but that doesn't help us in this PR. We should just support all 6 variations, it's simple.

Reply:
See #18237.